src/components/pages/pages.test.js

```diff
@@ -0,0 +1,239 @@
+  const [pageIndex, setPageIndex] = React.useState(
+    Number(props.initialpageIndex) ? Number(props.initialpageIndex) : undefined
+  );
+  const [isDisabled, setIsDisabled] = React.useState(false);
+
+  const handleCancel = () => {
+    setIsOpen(false);
+    setPageIndex(0);
+  };
+
+  const handleOpen = () => {
+    setIsOpen(true);
+
+    if (!props.initialpageIndex) {
+      setPageIndex(0);
+    } else setPageIndex(Number(props.initialpageIndex));
+  };
+
+  const handleOnClick = () => {
+    setIsDisabled(true);
+    setPageIndex(pageIndex + 1);
+    setTimeout(() => {
+      setIsDisabled(false);
+    }, 50);
+  };
+
+  const handleBackClick = (ev) => {
+    setIsDisabled(true);
+    setTimeout(() => {
+      setIsDisabled(false);
+    }, 50);
+
+    if (!isDisabled) {
+      ev.preventDefault();
+      setPageIndex(pageIndex - 1);
+    }
+  };
+
+  return (
+    <div>
+      <Button onClick={handleOpen}>Open Preview</Button>
+      <DialogFullScreen pagesStyling open={isOpen} onCancel={handleCancel}>
+        <Pages pageIndex={pageIndex} {...props}>
+          <Page title={<Heading title="My First Page" />}>
+            <Button onClick={handleOnClick} disabled={isDisabled}>
+              Go to second page
+            </Button>
+          </Page>
+          <Page
+            title={
+              <Heading title="My Second Page" backLink={handleBackClick} />
+            }
+          >
+            <Button onClick={handleOnClick} disabled={isDisabled}>
+              Go to third page
+            </Button>
+          </Page>
+          <Page
+            title={<Heading title="My Third Page" backLink={handleBackClick} />}
+          >
+            Third Page
+          </Page>
+        </Pages>
+      </DialogFullScreen>
+    </div>
+  );
+};
+
+const PageComponent = ({ children, ...props }) => {
+  return (
+    <Page title={<Heading title="My First Page" />} {...props}>
+      {children}
+    </Page>
+  );
+};
+
+context("Testing Pages component", () => {
+  describe("should render Pages component", () => {
+    // raised an issue FE-5491
+    it.skip("should render Pages component with className prop", () => {
+      CypressMountWithProviders(
+        <PagesComponent className={CHARACTERS.STANDARD} />
+      );
+
+      dataComponentButtonByText("Open Preview").click();
+      carousel().should("have.attr", "class").contains(CHARACTERS.STANDARD);
+    });
+
+    it.each([
+      [1, "Second"],
+      ["2", "Third"],
+    ])(
+      "should render Pages component with initialpageIndex set to %s",
+      (initialpageIndex, pageName) => {
+        CypressMountWithProviders(
+          <PagesComponent initialpageIndex={initialpageIndex} />
+        );
+
+        dataComponentButtonByText("Open Preview").click();
+        getDataElementByValue("title").should(
+          "have.text",
+          `My ${pageName} Page`
+        );
+      }
+    );
+
+    it.each([
+      [1, "Second"],
```

Comment:
Hi @ZhuoyuJin, thank you for your comment, but actually, we can't skip the value, because we are checking the possibility to pass `number` or `string` as `pageIndex` prop

Reply:
that should be in a `describe` block that encapsulates both cases